projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js

```diff
@@ -73,64 +118,119 @@ export const settings = {
 			! dontShowAgain &&
 			isModalOpen;
 
-		return (
-			showModal && (
-				<Modal
-					isDismissible={ true }
-					className="launchpad__save-modal"
-					onRequestClose={ () => {
-						// bypass the onRequestClose function the first time it's called when you publish a post because it closes the modal immediately
-						if ( isInitialPostPublish ) {
-							setIsInitialPostPublish( false );
-							return;
-						}
-						setIsModalOpen( false );
-						setDontShowAgain( isChecked );
-						recordTracksEvent( 'jetpack_launchpad_save_modal_close' );
-					} }
-				>
-					<div className="launchpad__save-modal-body">
-						<div className="launchpad__save-modal-text">
-							<h1 className="launchpad__save-modal-heading">
-								{ __( 'Great progress!', 'jetpack' ) }
-							</h1>
-							<p className="launchpad__save-modal-message">
-								{ __(
-									'You are one step away from bringing your site to life. Check out the next steps that will help you to launch your site.',
-									'jetpack'
-								) }
-							</p>
+		const firstPostPublishedModal = (
```

Comment:
I duplicate JSX markup between the first post published modal and the launchpad modal.

It's a bit unwieldy, but it makes abstraction way, way easier in the future, especially because I'm betting other folks will want to add their own modals here eventually.